src/varbc_gsi/GsiAircraftBiasConverter.cc

```diff
@@ -0,0 +1,130 @@
+ *     * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ *      */
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <iostream>
+
+#include <Eigen/Dense>
+
+#include "eckit/config/LocalConfiguration.h"
+#include "eckit/config/YAMLConfiguration.h"
+#include "eckit/exception/Exceptions.h"
+#include "eckit/filesystem/PathName.h"
+
+#include "ioda/ObsGroup.h"
+#include "ioda/Engines/HH.h"
+
+#include "oops/util/missingValues.h"
+
+#include "GsiAircraftBiasReader.h"
+
+ioda::ObsGroup makeObsBiasObject(ioda::Group &empty_base_object,
+                                 const std::string & coeffile,
+                                 const std::vector<std::string> & tailIds,
+                                 const std::vector<int> & lastCycleUpdated,
+                                 const std::vector<std::string> & predictors) {
+  /// Predictors
+  int numPreds = predictors.size();
+  int numIds = tailIds.size();
+
+  /// Creating dimensions: n
+  ioda::NewDimensionScales_t newDims {
+      ioda::NewDimensionScale<int>("nvars", 1),
+      ioda::NewDimensionScale<int>("nrecs", numIds)
+  };
+
+  /// Construct an ObsGroup object, with 2 dimensions nrecs, nvars
+  ioda::ObsGroup ogrp = ioda::ObsGroup::generate(empty_base_object, newDims);
+
+  /// Create tail IDs and cycles variable
+  ioda::Variable tailIdsVar = ogrp.vars.createWithScales<std::string>("records",
+                                    {ogrp.vars["nrecs"]});
+  tailIdsVar.write(tailIds);
+
+  ioda::Variable lastCycleUpdatedVar = ogrp.vars.createWithScales<int>("datetimes",
+                                            {ogrp.vars["nrecs"]});
+  lastCycleUpdatedVar.write(lastCycleUpdated);
+
+  /// Create 2D bias coefficient variable
+  Eigen::ArrayXXf biascoeffs(numIds, numPreds*3);
+
+  readObsBiasCoefficients(coeffile, biascoeffs);
+
+  for (int i = 0; i < numPreds; ++i) {
+    /// Set up the creation parameters for the bias coefficients variable
+    ioda::VariableCreationParameters float_params;
+    float_params.chunk = true;               // allow chunking
+    float_params.compressWithGZIP();         // compress using gzip
+    float missing_value = util::missingValue(missing_value);
+    float_params.setFillValue<float>(missing_value);
+
+    // Access predictor coefficient values column and create variable
+    Eigen::ArrayXXf subVar = biascoeffs.col(i);
+
+    ioda::Variable biasVar = ogrp.vars.createWithScales<float>("biasCoefficients/"+predictors[i],
+                       {ogrp.vars["nvars"], ogrp.vars["nrecs"]}, float_params);
+    biasVar.writeWithEigenRegular(subVar);
+
+    // Access predictor observation error values column and create variable
+    Eigen::ArrayXXf subVarObError = biascoeffs.col(i+3);
+
+    ioda::Variable biasVarObError = ogrp.vars.createWithScales<float>(
+                                                "biasCoeffObErrors/"+predictors[i],
```

Comment:
For my understanding: what does the "predictor observation error" mean?

Reply:
Good question. @ADCollard any comment? Only the value for constant is even populated. See in the linked issue: "observation error sums for each predictor (currently only column 5 is populated)"

Reply:
I believe that column (sixth column in the aircraft bias coefficient file) has the number of observations for that tail number in the cycle, that would have contributed to the update of the coefficient.

Reply:
@gmao-msienkie hmm, that would actually make a lot more sense. What are the next two columns supposed to be then? I know the last 3 are the background errors.

Reply:
this naming of variable and comments is a bit odd.  It really is the predictor error the `Ob` in all these names is just to let us know it's Obs related.
`obSubVarError`,,,, `obBiasCoeffErrors` makes a bit more sense as these are again predictor or coefficient error not `ObErrors`

and lastly are you saying just for the aircraft bias coefficient file the predictor (aircraft ascent rate) does not have any error assigned?

Reply:
I think that you would need to check with @gmao-yzhu to be sure of the intent (or original intent) of the three columns here.  She had worked on a couple of different methods for the bias correction - this one with ascent and ascent-squared, and another one that calculated coefficients to be applied depending on phase of flight (ascending or descending) so there would have been counts for each type (level flight, ascending, or descending).  Eventually only the method with vvel and vvel-squared ended up being used in NCEP operations, so the other columns are not needed.  Our code at GMAO still has the reference to the 'aircraft_t_bc_pof' method as well as my 'aircraft_bc_ext' that was used in our MERRA2 analysis.

Reply:
Just chiming in to say that @gmao-msienkie 's explanation is correct as far as I know.   The zeroed columns are used when the three phases of flight are treated differently.  I do not have a strong opinion on whether to retain this capability.

My explanation here [here](https://github.com/JCSDA-internal/ioda-converters/issues/804#issuecomment-1212274832) was not entirely correct (and confusing), so I will amend that.

Reply:
@gmao-msienkie @ADCollard Your explanation is correct. To keep the number of data count columns and the number of predictors the same may allow us to have a little more flexibility in the future. 
@gmao-msienkie Could you please make sure that changes made in this PR can be applied to your 'aircraft_bc_ext' that was used in our MERRA2 analysis? 

@BenjaminRuston I also feel that obBiasCoeffErrors makes more sense.

Reply:
Just so I understand what you are proposing:
you want something like:
`numObsAssimilated/constant(nrecs, nvars)`
`numObsAssimilated/vertical_velocity(nrecs, nvars)`
`numObsAssimilated/vertical_velocity_squared(nrecs, nvars)`
?

Reply:
Is this why you are discussing about the number of data count columns?  
When you are using "constant" + "vertical_velocity" + "order" in yaml file, you consider the predictor number to be 3 or 2?

Reply:
For what it's worth, unless you have a strong preference, I'd rather not have different number of obs assimilated for each predictor. I would advocate for:
`numObsAssimilated(nrecs, nvars)`
where in the aircraft case, `nrecs` = ntailnumbers, and `nvars`=1=`airTemperature` (for now)
and in the radiance case, `nrecs`=1 and `nvars`=`nchans`
because I don't particularly want to clutter the radiance files with a bunch of new groups, and we would need to be consistent across both radiance and aircraft VarBC files.

Reply:
If consistency is your priority, keeping only one slot for observation number is fine with me. 

Since you mentioned radiance bias correction, for all-sky radiance, I would strongly suggest to use the observation number that are actually used in the radiance bias correction calculation, not the observation number that are assimilated.

Reply:
Good point, perhaps it should be something like `numObsUsed`?